test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/DefaultHtmlGeneratorTest.cs

```diff
@@ -651,6 +652,26 @@ public void GetCurrentValues_WithModel_ReturnsModel(string[] rawValue, IReadOnly
             Assert.Equal(expectedAntiforgeryHtmlField, antiforgeryField);
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void GenerateAntiforgery_AlwaysGeneratesAntiforgeryToken_IfCannotRenderAtEnd(bool hasAntiforgeryToken)
```

Comment:
This is absolutely a valid scenario and it is _not_ about being outside a `<form>` element. Instead think of cases such as the following when tag helpers are not enabled.
``` html
<form action="/Home/Create">
    @Html.AntiForgeryToken()
</form>
```